File: ios/Sasquatsh/Views/Toolbox/RandomGamePickerView.swift
import SwiftUI

struct PickedGame {
    let name: String
    let yearPublished: Int?
    let thumbnailUrl: String?
    let minPlayers: Int?
    let maxPlayers: Int?
    let bggRank: Int?

    init(cached: BggCachedGame) {
        name = cached.name
        yearPublished = cached.yearPublished
        thumbnailUrl = cached.thumbnailUrl
        minPlayers = cached.minPlayers
        maxPlayers = cached.maxPlayers
        bggRank = cached.bggRank
    }

    init(collection: CollectionGame) {
        name = collection.gameName
        yearPublished = collection.yearPublished
        thumbnailUrl = collection.thumbnailUrl
        minPlayers = collection.minPlayers
        maxPlayers = collection.maxPlayers
        bggRank = collection.bggRank
    }
}

enum GamePickerMode: String, CaseIterable {
    case customPick = "Custom"
    case myCollection = "My Collection"
    case top50 = "BGG Top 50"
    case random = "Random"
}

struct RandomGamePickerView: View {
    @Environment(\.dismiss) private var dismiss
    @Environment(\.services) private var services
    @State private var mode: GamePickerMode = .customPick
    @State private var selectedGameIds: Set<String> = []
    @State private var playerCount: Int = 4
    @State private var allGames: [BggCachedGame] = []
    @State private var myCollection: [CollectionGame] = []
    @State private var pickedGame: PickedGame?
    @State private var isLoading = false
    @State private var isLoadingGames = false
    @State private var error: String?
    @State private var showResult = false
    @State private var hasLoadedCache = false
    @State private var hasLoadedCollection = false

    var body: some View {
        VStack(spacing: 0) {
            CompactNavBar(title: "Random Game Picker") { dismiss() }
            ScrollViewReader { proxy in
            ScrollView {
                VStack(spacing: 16) {
                    // Player count — always visible
                    HStack(spacing: 14) {
                        Image(systemName: "person.2")
                            .font(.system(size: 16))
                            .foregroundStyle(Color.md3Primary)

                        Text("Min. Players")
                            .font(.md3BodyMedium)
                            .foregroundStyle(Color.md3OnSurfaceVariant)

                        Spacer()

                        HStack(spacing: 12) {
                            Button {
                                if playerCount > 1 { playerCount -= 1 }
                            } label: {
                                Image(systemName: "minus.circle.fill")
                                    .font(.system(size: 26))
                                    .foregroundStyle(playerCount > 1 ? Color.md3Primary : Color.md3OnSurfaceVariant.opacity(0.3))
                            }
                            .disabled(playerCount <= 1)

                            Text("\(playerCount)")
                                .font(.system(size: 28, weight: .bold, design: .rounded))
                                .foregroundStyle(Color.md3OnSurface)
                                .frame(width: 40)

                            Button {
                                if playerCount < 20 { playerCount += 1 }
                            } label: {
                                Image(systemName: "plus.circle.fill")
                                    .font(.system(size: 26))
                                    .foregroundStyle(playerCount < 20 ? Color.md3Primary : Color.md3OnSurfaceVariant.opacity(0.3))
                            }
                            .disabled(playerCount >= 20)
                        }
                    }
                    .padding(14)
                    .background(Color.md3Surface)
                    .clipShape(RoundedRectangle(cornerRadius: MD3Shape.medium))
                    .padding(.horizontal, 20)
                    .padding(.top, 8)

                    // Mode picker
                    VStack(alignment: .leading, spacing: 6) {
                        Text("From:")
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)

                        ScrollView(.horizontal, showsIndicators: false) {
                            HStack(spacing: 8) {
                                ForEach(GamePickerMode.allCases, id: \.self) { m in
                                    Button {
                                        withAnimation { mode = m }
                                    } label: {
                                        HStack(spacing: 4) {
                                            if mode == m {
                                                Image(systemName: "checkmark")
                                                    .font(.system(size: 11, weight: .semibold))
                                            }
                                            Text(m.rawValue)
                                                .font(.md3LabelLarge)
                                        }
                                        .padding(.horizontal, 16)
                                        .frame(height: 32)
                                        .background(mode == m ? Color.md3SecondaryContainer : Color.clear)
                                        .foregroundStyle(mode == m ? Color.md3OnSecondaryContainer : Color.md3OnSurfaceVariant)
                                        .clipShape(Capsule())
                                        .overlay(
                                            Capsule().stroke(mode == m ? Color.clear : Color.md3Outline, lineWidth: 1)
                                        )
                                    }
                                }
                            }
                        }
                    }
                    .padding(.horizontal, 20)

                    // Result — show above the game list
                    if showResult, let game = pickedGame {
                        gameResultCard(game)
                            .transition(.opacity.combined(with: .scale(scale: 0.9)))
                            .padding(.horizontal, 20)
                            .id("result")
                    } else if showResult && pickedGame == nil {
                        VStack(spacing: 8) {
                            Image(systemName: "exclamationmark.triangle")
                                .font(.system(size: 32))
                                .foregroundStyle(Color.md3Error.opacity(0.6))
                            Text("No games found matching your criteria")
                                .font(.md3BodyMedium)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                                .multilineTextAlignment(.center)
                        }
                        .padding(.top, 16)
                    }

                    // Custom pick — selectable game list
                    if mode == .customPick && hasLoadedCollection && !myCollection.isEmpty {
                        VStack(alignment: .leading, spacing: 8) {
                            HStack {
                                Text("Select games to pick from")
                                    .font(.md3BodySmall)
                                    .foregroundStyle(Color.md3OnSurfaceVariant)
                                Spacer()
                                if !selectedGameIds.isEmpty {
                                    Text("\(selectedGameIds.count) selected")
                                        .font(.system(size: 12, weight: .semibold))
                                        .foregroundStyle(Color.md3Primary)
                                }
                            }

                            ForEach(myCollection) { game in
                                let isSelected = selectedGameIds.contains(game.id)
                                Button {
                                    if isSelected {
                                        selectedGameIds.remove(game.id)
                                    } else {
                                        selectedGameIds.insert(game.id)
                                    }
                                } label: {
                                    HStack(spacing: 10) {
                                        Image(systemName: isSelected ? "checkmark.circle.fill" : "circle")
                                            .font(.system(size: 20))
                                            .foregroundStyle(isSelected ? Color.md3Primary : Color.md3OnSurfaceVariant.opacity(0.4))

                                        if let urlStr = game.thumbnailUrl, let url = URL(string: urlStr) {
                                            AsyncImage(url: url) { image in
                                                image.resizable().aspectRatio(contentMode: .fill)
                                            } placeholder: {
                                                Color.md3SurfaceVariant
                                            }
                                            .frame(width: 36, height: 36)
                                            .clipShape(RoundedRectangle(cornerRadius: 4))
                                        }

                                        VStack(alignment: .leading, spacing: 2) {
                                            Text(game.gameName)
                                                .font(.system(size: 14, weight: .medium))
                                                .foregroundStyle(Color.md3OnSurface)
                                                .lineLimit(1)
                                            if let min = game.minPlayers, let max = game.maxPlayers {
                                                Text(min == max ? "\(min) players" : "\(min)–\(max) players")
                                                    .font(.system(size: 11))
                                                    .foregroundStyle(Color.md3OnSurfaceVariant)
                                            }
                                        }

                                        Spacer()
                                    }
                                    .padding(.vertical, 6)
                                    .padding(.horizontal, 10)
                                    .background(isSelected ? Color.md3PrimaryContainer.opacity(0.3) : Color.clear)
                                    .clipShape(RoundedRectangle(cornerRadius: MD3Shape.small))
                                }
                                .buttonStyle(.plain)
                            }
                        }
                        .padding(.horizontal, 20)
                    } else if mode == .customPick && hasLoadedCollection && myCollection.isEmpty {
                        VStack(spacing: 8) {
                            Image(systemName: "tray")
                                .font(.system(size: 32))
                                .foregroundStyle(Color.md3OnSurfaceVariant.opacity(0.4))
                            Text("Your collection is empty")
                                .font(.md3BodyMedium)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                            Text("Add games to your collection on the website")
                                .font(.md3BodySmall)
                                .foregroundStyle(Color.md3OnSurfaceVariant.opacity(0.7))
                        }
                        .padding(.top, 16)
                    }

                    // Loading cache indicator
                    if isLoadingGames {
                        HStack(spacing: 8) {
                            ProgressView()
                                .tint(Color.md3Primary)
                            Text("Loading game library...")
                                .font(.md3BodySmall)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                        }
                        .padding(.top, 8)
                    }

                    // Empty collection states for non-custom modes
                    if mode == .myCollection && hasLoadedCollection && myCollection.isEmpty {
                        VStack(spacing: 8) {
                            Image(systemName: "tray")
                                .font(.system(size: 32))
                                .foregroundStyle(Color.md3OnSurfaceVariant.opacity(0.4))
                            Text("Your collection is empty")
                                .font(.md3BodyMedium)
                                .foregroundStyle(Color.md3OnSurfaceVariant)
                            Text("Add games to your collection on the website")
                                .font(.md3BodySmall)
                                .foregroundStyle(Color.md3OnSurfaceVariant.opacity(0.7))
                        }
                        .padding(.top, 16)
                    }

                    if isLoading {
                        ProgressView()
                            .tint(Color.md3Primary)
                            .scaleEffect(1.3)
                            .padding(.top, 16)
                    }

                    if let error {
                        Text(error)
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3Error)
                            .padding(.horizontal, 20)
                    }
                }
                .padding(.bottom, 16)
            }
            .onChange(of: showResult) { _, showing in
                if showing {
                    withAnimation {
                        proxy.scrollTo("result", anchor: .top)
                    }
                }
            }
            } // ScrollViewReader

            // Action button
            let isDisabled = isLoading || isLoadingGames || (mode == .customPick && selectedGameIds.isEmpty)
            Button {
                pickGame()
            } label: {
                HStack(spacing: 8) {
                    Image(systemName: showResult ? "arrow.clockwise" : "dice.fill")
                    Text(showResult ? "Pick Another" : "Pick a Game!")
                }
                .primaryButtonStyle()
                .opacity(isDisabled ? 0.4 : 1.0)
            }
            .disabled(isDisabled)
            .padding(.horizontal, 20)
            .padding(.vertical, 12)
            .background(Color.md3SurfaceContainer)
        }
        .background(Color.md3SurfaceContainer)
        .toolbar(.hidden, for: .navigationBar)
        .task {
            await loadData()
        }
    }

    // MARK: - Game Result Card

    private func gameResultCard(_ game: PickedGame) -> some View {
        VStack(spacing: 0) {
            // Thumbnail
            if let urlStr = game.thumbnailUrl, let url = URL(string: urlStr) {
                AsyncImage(url: url) { image in
                    image.resizable().aspectRatio(contentMode: .fit)
                } placeholder: {
                    Color.md3SurfaceVariant
                        .frame(height: 160)
                        .overlay {
                            ProgressView()
                                .tint(Color.md3Primary)
                        }
                }
                .frame(maxHeight: 200)
                .clipShape(RoundedRectangle(cornerRadius: MD3Shape.medium, style: .continuous))
            }

            VStack(spacing: 10) {
                Text(game.name)
                    .font(.system(size: 20, weight: .bold))
                    .foregroundStyle(Color.md3OnSurface)
                    .multilineTextAlignment(.center)

                if let year = game.yearPublished {
                    Text("(\(String(year)))")
                        .font(.md3BodySmall)
                        .foregroundStyle(Color.md3OnSurfaceVariant)
                }

                HStack(spacing: 16) {
                    if let min = game.minPlayers, let max = game.maxPlayers {
                        Label {
                            Text(min == max ? "\(min) players" : "\(min)–\(max) players")
                                .font(.md3BodySmall)
                        } icon: {
                            Image(systemName: "person.2")
                                .foregroundStyle(Color.md3Primary)
                        }
                    }

                    if let rank = game.bggRank, rank > 0 {
                        Label {
                            Text("#\(rank) on BGG")
                                .font(.md3BodySmall)
                        } icon: {
                            Image(systemName: "trophy")
                                .foregroundStyle(Color.md3Tertiary)
                        }
                    }
                }
                .foregroundStyle(Color.md3OnSurfaceVariant)
            }
            .padding(16)
        }
        .background(Color.md3Surface)
        .clipShape(RoundedRectangle(cornerRadius: MD3Shape.large))
        .shadow(color: .black.opacity(0.08), radius: 6, y: 3)
    }

    // MARK: - Logic

    private func loadData() async {
        isLoadingGames = true

        // Load collection and cache in parallel
        async let collectionTask: () = loadCollection()
        async let cacheTask: () = loadGameCache()
        _ = await (collectionTask, cacheTask)

        isLoadingGames = false
    }

    private func loadCollection() async {
        guard !hasLoadedCollection else { return }
        do {
            myCollection = try await services.collections.getMyCollection()
            hasLoadedCollection = true
        } catch {
            print("Collection load error: \(error)")
            hasLoadedCollection = true // Mark loaded so we show empty state
        }
    }

    private func loadGameCache() async {
        guard !hasLoadedCache else { return }
        do {
            var page = 1
            var allFetched: [BggCachedGame] = []
            var hasMore = true
            while hasMore {
                let response = try await services.bgg.listCachedGames(page: page, limit: 100)
                allFetched.append(contentsOf: response.games)
                hasMore = page < response.totalPages
                page += 1
                if page > 10 { break }
            }
            allGames = allFetched
            hasLoadedCache = true
        } catch {
            self.error = "Failed to load game library"
        }
    }

    private func pickGame() {
        isLoading = true
        error = nil
        withAnimation {
            showResult = false
            pickedGame = nil
        }

        Task {
            try? await Task.sleep(for: .milliseconds(400))

            let picked: PickedGame?

            switch mode {
            case .customPick:
                let selected = myCollection.filter { selectedGameIds.contains($0.id) }
                picked = selected.randomElement().map { PickedGame(collection: $0) }

            case .myCollection:
                let filtered = myCollection.filter { game in
                    guard let max = game.maxPlayers else { return true }
                    return playerCount <= max
                }
                picked = filtered.randomElement().map { PickedGame(collection: $0) }

            case .top50:
                let filtered = allGames
                    .filter { game in
                        guard let max = game.maxPlayers else { return true }
                        return playerCount <= max
                    }
                    .filter { $0.bggRank != nil && $0.bggRank! > 0 }
                    .sorted { ($0.bggRank ?? Int.max) < ($1.bggRank ?? Int.max) }
                    .prefix(50)
                picked = Array(filtered).randomElement().map { PickedGame(cached: $0) }

            case .random:
                let filtered = allGames.filter { game in
                    guard let max = game.maxPlayers else { return true }
                    return playerCount <= max
                }
                picked = filtered.randomElement().map { PickedGame(cached: $0) }
            }

            withAnimation(.spring(response: 0.5, dampingFraction: 0.7)) {
                pickedGame = picked
                showResult = true
                isLoading = false
            }
        }
    }

}

File: android/app/src/main/java/com/sasquatsh/app/views/toolbox/RandomGamePickerView.kt
package com.sasquatsh.app.views.toolbox

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.fadeIn
import androidx.compose.animation.scaleIn
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Casino
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.Group
import androidx.compose.material.icons.filled.Inbox
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Remove
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material.icons.outlined.Circle
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import coil.compose.AsyncImage
import com.sasquatsh.app.models.BggCachedGame
import com.sasquatsh.app.models.CollectionGame
import com.sasquatsh.app.viewmodels.CollectionViewModel

private data class PickedGame(
    val name: String,
    val yearPublished: Int? = null,
    val thumbnailUrl: String? = null,
    val minPlayers: Int? = null,
    val maxPlayers: Int? = null,
    val bggRank: Int? = null
)

private enum class GamePickerMode(val label: String) {
    MY_COLLECTION("My Collection"),
    CUSTOM("Custom"),
    RANDOM("Random")
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RandomGamePickerView(
    viewModel: CollectionViewModel = hiltViewModel()
) {
    val uiState by viewModel.uiState.collectAsState()
    var playerCount by rememberSaveable { mutableIntStateOf(4) }
    var mode by rememberSaveable { mutableStateOf(GamePickerMode.MY_COLLECTION) }
    var pickedGame by remember { mutableStateOf<PickedGame?>(null) }
    var showResult by remember { mutableStateOf(false) }
    var isPicking by remember { mutableStateOf(false) }
    val selectedGameIds = remember { mutableSetOf<String>() }

    LaunchedEffect(Unit) {
        viewModel.loadCollection()
    }

    Scaffold(
        topBar = { TopAppBar(title = { Text("Random Game Picker") }) }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
        ) {
            LazyColumn(
                contentPadding = PaddingValues(vertical = 8.dp),
                verticalArrangement = Arrangement.spacedBy(16.dp),
                modifier = Modifier.weight(1f)
            ) {
                // Player count
                item {
                    Card(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = 20.dp),
                        colors = CardDefaults.cardColors(
                            containerColor = MaterialTheme.colorScheme.surface
                        )
                    ) {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(14.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Row(verticalAlignment = Alignment.CenterVertically) {
                                Icon(
                                    Icons.Default.Group,
                                    contentDescription = null,
                                    tint = MaterialTheme.colorScheme.primary,
                                    modifier = Modifier.size(16.dp)
                                )
                                Spacer(modifier = Modifier.width(8.dp))
                                Text("Min. Players", style = MaterialTheme.typography.bodyMedium)
                            }

                            Row(
                                horizontalArrangement = Arrangement.spacedBy(12.dp),
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                IconButton(
                                    onClick = { if (playerCount > 1) playerCount-- },
                                    enabled = playerCount > 1
                                ) {
                                    Icon(Icons.Default.Remove, contentDescription = "Decrease")
                                }
                                Text(
                                    "$playerCount",
                                    fontSize = 28.sp,
                                    fontWeight = FontWeight.Bold
                                )
                                IconButton(
                                    onClick = { if (playerCount < 20) playerCount++ },
                                    enabled = playerCount < 20
                                ) {
                                    Icon(Icons.Default.Add, contentDescription = "Increase")
                                }
                            }
                        }
                    }
                }

                // Mode picker
                item {
                    Column(modifier = Modifier.padding(horizontal = 20.dp)) {
                        Text(
                            "From:",
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Spacer(modifier = Modifier.height(6.dp))
                        Row(
                            modifier = Modifier.horizontalScroll(rememberScrollState()),
                            horizontalArrangement = Arrangement.spacedBy(8.dp)
                        ) {
                            GamePickerMode.values().forEach { m ->
                                FilterChip(
                                    selected = mode == m,
                                    onClick = { mode = m },
                                    label = { Text(m.label) },
                                    leadingIcon = if (mode == m) ({
                                        Icon(
                                            Icons.Default.CheckCircle,
                                            contentDescription = null,
                                            modifier = Modifier.size(16.dp)
                                        )
                                    }) else null
                                )
                            }
                        }
                    }
                }

                // Result
                if (showResult && pickedGame != null) {
                    item {
                        AnimatedVisibility(
                            visible = showResult,
                            enter = fadeIn() + scaleIn(initialScale = 0.9f)
                        ) {
                            GameResultCard(pickedGame!!)
                        }
                    }
                } else if (showResult && pickedGame == null) {
                    item {
                        Column(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(top = 16.dp),
                            horizontalAlignment = Alignment.CenterHorizontally
                        ) {
                            Icon(
                                Icons.Default.Warning,
                                contentDescription = null,
                                modifier = Modifier.size(32.dp),
                                tint = MaterialTheme.colorScheme.error.copy(alpha = 0.6f)
                            )
                            Spacer(modifier = Modifier.height(8.dp))
                            Text(
                                "No games found matching your criteria",
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurfaceVariant,
                                textAlign = TextAlign.Center
                            )
                        }
                    }
                }

                // Custom pick: show collection for selection
                if (mode == GamePickerMode.CUSTOM && uiState.myGames.isNotEmpty()) {
                    item {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 20.dp),
                            horizontalArrangement = Arrangement.SpaceBetween
                        ) {
                            Text(
                                "Select games to pick from",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                            if (selectedGameIds.isNotEmpty()) {
                                Text(
                                    "${selectedGameIds.size} selected",
                                    fontSize = 12.sp,
                                    fontWeight = FontWeight.SemiBold,
                                    color = MaterialTheme.colorScheme.primary
                                )
                            }
                        }
                    }
                    items(uiState.myGames, key = { it.id }) { game ->
                        val isSelected = selectedGameIds.contains(game.id)
                        Card(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 20.dp)
                                .clickable {
                                    if (isSelected) selectedGameIds.remove(game.id)
                                    else selectedGameIds.add(game.id)
                                },
                            colors = CardDefaults.cardColors(
                                containerColor = if (isSelected)
                                    MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.3f)
                                else MaterialTheme.colorScheme.surface
                            )
                        ) {
                            Row(
                                modifier = Modifier.padding(horizontal = 10.dp, vertical = 6.dp),
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Icon(
                                    if (isSelected) Icons.Default.CheckCircle else Icons.Outlined.Circle,
                                    contentDescription = null,
                                    tint = if (isSelected) MaterialTheme.colorScheme.primary
                                    else MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.4f),
                                    modifier = Modifier.size(20.dp)
                                )
                                Spacer(modifier = Modifier.width(10.dp))
                                game.thumbnailUrl?.let { url ->
                                    AsyncImage(
                                        model = url,
                                        contentDescription = null,
                                        modifier = Modifier
                                            .size(36.dp)
                                            .clip(RoundedCornerShape(4.dp)),
                                        contentScale = ContentScale.Crop
                                    )
                                    Spacer(modifier = Modifier.width(10.dp))
                                }
                                Column(modifier = Modifier.weight(1f)) {
                                    Text(
                                        game.gameName,
                                        fontSize = 14.sp,
                                        fontWeight = FontWeight.Medium,
                                        maxLines = 1
                                    )
                                    if (game.minPlayers != null && game.maxPlayers != null) {
                                        Text(
                                            "${game.minPlayers}-${game.maxPlayers} players",
                                            fontSize = 11.sp,
                                            color = MaterialTheme.colorScheme.onSurfaceVariant
                                        )
                                    }
                                }
                            }
                        }
                    }
                }

                // Empty states
                if (mode == GamePickerMode.MY_COLLECTION && !uiState.isLoading && uiState.myGames.isEmpty()) {
                    item {
                        Column(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(top = 16.dp),
                            horizontalAlignment = Alignment.CenterHorizontally
                        ) {
                            Icon(
                                Icons.Default.Inbox,
                                contentDescription = null,
                                modifier = Modifier.size(32.dp),
                                tint = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.4f)
                            )
                            Text(
                                "Your collection is empty",
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                }

                if (uiState.isLoading) {
                    item {
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.Center
                        ) {
                            CircularProgressIndicator(modifier = Modifier.size(32.dp))
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(
                                "Loading game library...",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                }

                if (isPicking) {
                    item {
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.Center
                        ) {
                            CircularProgressIndicator(modifier = Modifier.size(32.dp))
                        }
                    }
                }
            }

            // Action button
            val isDisabled = isPicking || uiState.isLoading ||
                    (mode == GamePickerMode.CUSTOM && selectedGameIds.isEmpty())
            Button(
                onClick = {
                    isPicking = true
                    showResult = false
                    pickedGame = null

                    val games = uiState.myGames
                    val picked: PickedGame? = when (mode) {
                        GamePickerMode.CUSTOM -> {
                            val selected = games.filter { selectedGameIds.contains(it.id) }
                            selected.randomOrNull()?.let { g ->
                                PickedGame(g.gameName, g.yearPublished, g.thumbnailUrl, g.minPlayers, g.maxPlayers, g.bggRank)
                            }
                        }
                        GamePickerMode.MY_COLLECTION -> {
                            val filtered = games.filter { g ->
                                val max = g.maxPlayers ?: return@filter true
                                playerCount <= max
                            }
                            filtered.randomOrNull()?.let { g ->
                                PickedGame(g.gameName, g.yearPublished, g.thumbnailUrl, g.minPlayers, g.maxPlayers, g.bggRank)
                            }
                        }
                        GamePickerMode.RANDOM -> {
                            val filtered = games.filter { g ->
                                val max = g.maxPlayers ?: return@filter true
                                playerCount <= max
                            }
                            filtered.randomOrNull()?.let { g ->
                                PickedGame(g.gameName, g.yearPublished, g.thumbnailUrl, g.minPlayers, g.maxPlayers, g.bggRank)
                            }
                        }
                    }
                    pickedGame = picked
                    showResult = true
                    isPicking = false
                },
                enabled = !isDisabled,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 12.dp)
            ) {
                Icon(
                    if (showResult) Icons.Default.Refresh else Icons.Default.Casino,
                    contentDescription = null,
                    modifier = Modifier.size(18.dp)
                )
                Spacer(modifier = Modifier.width(8.dp))
                Text(if (showResult) "Pick Another" else "Pick a Game!")
            }
        }
    }
}

@Composable
private fun GameResultCard(game: PickedGame) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 20.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        shape = RoundedCornerShape(16.dp)
    ) {
        Column {
            game.thumbnailUrl?.let { url ->
                AsyncImage(
                    model = url,
                    contentDescription = game.name,
                    modifier = Modifier
                        .fillMaxWidth()
                        .heightIn(max = 200.dp)
                        .clip(RoundedCornerShape(topStart = 16.dp, topEnd = 16.dp)),
                    contentScale = ContentScale.Fit
                )
            }

            Column(
                modifier = Modifier.padding(16.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Text(
                    game.name,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    textAlign = TextAlign.Center
                )
                game.yearPublished?.let {
                    Text(
                        "($it)",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
                Spacer(modifier = Modifier.height(8.dp))
                Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                    if (game.minPlayers != null && game.maxPlayers != null) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                Icons.Default.Group,
                                contentDescription = null,
                                modifier = Modifier.size(14.dp),
                                tint = MaterialTheme.colorScheme.primary
                            )
                            Spacer(modifier = Modifier.width(4.dp))
                            Text(
                                "${game.minPlayers}-${game.maxPlayers} players",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                    if (game.bggRank != null && game.bggRank > 0) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(
                                Icons.Default.EmojiEvents,
                                contentDescription = null,
                                modifier = Modifier.size(14.dp),
                                tint = MaterialTheme.colorScheme.tertiary
                            )
                            Spacer(modifier = Modifier.width(4.dp))
                            Text(
                                "#${game.bggRank} on BGG",
                                style = MaterialTheme.typography.bodySmall,
                                color = MaterialTheme.colorScheme.onSurfaceVariant
                            )
                        }
                    }
                }
            }
        }
    }
}
